Tests/SwiftParserTest/Assertions.swift

```diff
@@ -324,7 +324,7 @@ func assertNote(
   XCTAssertEqual(note.message, spec.message, file: spec.file, line: spec.line)
   let locationConverter = SourceLocationConverter(fileName: "", tree: tree)
   assertLocation(
-    note.location(converter: locationConverter),
+    note.location(converter: locationConverter)!,
```

Comment:
We should make sure that the test doesn’t crash if a note without a source location should be generated. Instead, we should raise a test failure.